contest/contest_info.py

```diff
@@ -175,6 +178,10 @@ def can_delete_contest(user, contest):
 HAS_ATTENDED = "has_attended"
 IS_ADMIN = "is_admin"
 OK = "OK"
+
+'''
+return error or success message
+'''
 def can_register_return_status(user, contest):
     if not user.is_authenticated():
```

Comment:
I noticed that in your template, the register button will not appear when user not login. (You checked it in ```show_register_btn``` filter)
Moreover, I traced where ```can_register_return_status``` function is called, below is the structure:
```
'can_register_return_status' called by
    'can_register_log' called by
        'register_user' called by
            'register_group', 'register_anonymous', and 'register'
    'can_register' called by
        'register_anonymous', and 'show_register_btn'
```
All of involved functions are either protected by ```@login_required``` or  check whether the users login first.
In other word, this if statement will never be touched, the user parameter will never be an unauthenticated user.